Packs/IOCParser/Integrations/IOCParser/IOCParser.py

```diff
@@ -284,41 +285,101 @@ def ioc_from_url_command(client: Client, args: Dict[str, Any]) -> List[CommandRe
     """
-        Returns the results of the Parse IOCs from text API call
+        Returns the results of the Parse IOCs from text API call.
         Args:
-            client: IOCParser client to use
-            args: All command arguments, text, limit and keys (if specified)
+            client: IOCParser client to use.
+            args: All command arguments, text, limit, and keys (if specified).
 
         Returns:
             CommandResults object containing the results of the parse from
-            text (in JSON format) as returned from the API and its readable output
+            text (in JSON format) as returned from the API and its readable output.
+    """
 
-        """
     data = args.get('data')
-    keys = argToList(args.get('keys'))
-    limit = args.get('limit')
-    if not keys:
-        keys = KEYS
-    keys = list_to_upper_case(keys)
+    try:
+        json.loads(str(data))
+    except ValueError as e:
+        raise ValueError(f'Data should be in JSON format. Error: {str(e)}')
+    keys = list_to_upper_case(argToList(args.get('keys'))) or KEYS
+    limit = arg_to_number(args.get('limit'))
 
-    if not data:
-        raise ValueError('text not specified')
+    try:
+        response = client.ioc_from_json_text(data)
+    except DemistoException as e:
+        raise ValueError(str(e))
+    response_data = remove_unwanted_data_from_response(response, keys, limit)
+    if not response_data:
+        raise ValueError('There is no information about the requested keys')
+    command_results = []
+    outputs: Dict[str, Any] = {'data': data, 'Results': []}
+    for ioc_type, iocs in response_data.items():
+        for ioc in iocs:
+            outputs.setdefault('Results', []).append({'type': ioc_type, 'value': ioc})
+        command_results.append(CommandResults(
+            readable_output=tableToMarkdown(f'results for {ioc_type}', iocs, headers=ioc_type),
+            outputs_prefix='IOCParser.parseFromJSONText',
+            outputs=outputs
+        ))
+
+    command_results.append(CommandResults(
+        raw_response=response_data
+    ))
+
+    return command_results
 
-    response = client.ioc_from_text(data)
 
-    response_data = process_response(response, keys, limit)
+def ioc_from_raw_text_command(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
+    """
+        Returns the results of the Parse IOCs from raw text API call
+        Args:
+            client: IOCParser client to use
+            args: All command arguments, text, limit, and keys (if specified).
+
+        Returns:
+            CommandResults object containing the results of the parse from
+            text (in JSON format) as returned from the API and its readable output
+    """
 
+    raw_text = args.get('data')
+    file = args.get('entry_id')
+    if raw_text:
+        if file:
+            raise ValueError('Both data and entry id were inserted - please insert only one.')
```

Comment:
```suggestion
            raise DemistoException('Both data and entry id were inserted - please insert only one.')
```

Almost every time you raise an exception it's a DemistoException.
Look at the description of [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) - it's not the case in the code.